### PR TITLE
fix(plugin-auth-backend): upgrade passport-saml to remove security issue

### PR DESCRIPTION
**Pull request by jmfrancois**
## Hey, I just made a Pull Request!

Using backstage by default create a high security issue reported by github since #3863:

Screen Shot 2021-01-13 at 10 47 32

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Review by freben — APPROVED** [review]
Thanks!

File: plugins/auth-backend/config.d.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ export interface Config {
         cert?: string;
         privateKey?: string;
         decryptionPvk?: string;
-        signatureAlgorithm?: 'sha1' | 'sha256' | 'sha512';
+        signatureAlgorithm?: 'sha256' | 'sha512';
```
Comment (freben) [inline plugins/auth-backend/config.d.ts:54]: Let me just double check - is this strictly necessary, AFTER actually upgrading? I am not sure if anybody actually uses sha1, but if they do, and usage is safe in v2, maybe we should keep it

Comment (jmfrancois) [reply plugins/auth-backend/config.d.ts:54]: It is reported in their changelog. https://github.com/node-saml/passport-saml/releases/tag/v2.0.0
Who wan to use sha1 ...

Comment (freben) [reply plugins/auth-backend/config.d.ts:54]: Yeah agreed :) But you know, legacy situations or whatnot

Comment (jmfrancois) [reply plugins/auth-backend/config.d.ts:54]: I know people will complain after upgrade it do not work !

but if you upgrade your install you can pay the price to update the configuration because of security issue.

As you wish 🤷

**Comment by jmfrancois** [comment]
By the way It is my first PR here.
Can I rely on the CI, should I fix the fail checks ?

File: yarn.lock [hunk 2]
```diff
@@ -2436,9 +2436,11 @@
     to-fast-properties "^2.0.0"
 
 "@backstage/catalog-model@^0.2.0":
-  version "0.6.0"
+  version "0.2.0"
```
Comment (freben) [inline yarn.lock:2439]: unrelated changes snuck in again!

**Comment by freben** [comment]
yeah feel free to use the CI to check things for you, you can of course also run `yarn tsc` in your local root to find typescript errors before pushing, and running `yarn test` inside the affected package too

**Comment by freben** [comment]
Our goal is to have a rock solid, thorough, well maintained CI experience. If something is amiss, we will want to fix it.

File: yarn.lock (hunk 2 above)
Comment (jmfrancois) [reply yarn.lock]: Thanks this is fixed

File: plugins/auth-backend/package.json [hunk 3]
```diff
@@ -74,6 +74,8 @@
     "@types/passport-google-oauth20": "^2.0.3",
     "@types/passport-microsoft": "^0.0.0",
     "@types/passport-saml": "^1.1.2",
+    "@types/passport-strategy": "^0.2.35",
+    "@types/xml2js": "^0.4.7",
```
Comment (jmfrancois) [inline plugins/auth-backend/package.json:78]: I have added those two dev deps to remove some warnings/errors in the CI

**Comment by freben** [comment]
Great, thanks!
